### PR TITLE
Create `Nub` module (#68)

### PR DESCRIPTION
**Pull request by vrom911**
Here is how `Nub` documentation looks like at the moment:
image

### CODE REVIEW

File: src/Nub.hs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ordNub :: (Ord a) => [a] -> [a]
+ordNub = go Set.empty
+  where
+    go _ []     = []
+    go s (x:xs) =
+      if x `Set.member` s
+      then go s xs
+      else x : go (Set.insert x s) xs
+
+-- | Like 'Prelude.nub' but runs in @O(n * log_16(n))@ time and requires 'Hashable'.
```
Comment (ChShersh) [inline src/Nub.hs]: It would be really good to add examples in documentation to each function.

File: src/Nub.hs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ Here are recomendations of usage of particular functions based on benchmarking resutls.
+ Benchmarks were made on lists of 'Prelude.Int's and 'Data.Text.Text's.
+
+ * 'hashNub' is faster than 'ordNub' when there're not so many different values in the list.
+
+ * 'hashNub' is the fastest with 'Data.Text.Text'.
+
+ * 'sortNub' has better performance than 'ordNub' but should be used when sorting is also needed.
+
+ * 'unstableNub' has better performance than 'hashNub'.
```
Comment (ChShersh) [inline src/Nub.hs]: It would be good to add that fact that 'unstableNub' doesn't save the order here as well.

File: src/Nub.hs [hunk 3]
```diff
@@ -1,4 +1,25 @@
--- | Functions to remove dublicates from a list.
+{-| Functions to remove dublicates from a list.
```
Comment (ChShersh) [inline src/Nub.hs]: `duplicates` maybe? Is it a typo?

Comment (vrom911) [reply src/Nub.hs]: Ooops..

**Comment by vrom911** [comment]
@ChShersh I've changed the example so that it would be seen, that `unstableNub` doesn't sort the list as `sortNub` do..

**Comment by ChShersh** [comment]
Oh, @vrom911, sorry, I forgot one thing. Could you please aslo bump up version in `.cabal` file to `0.7.1` and update `CHANGELOG` as well? You can even update related parts of `README` :)

**Pull request merged**